client/js/table-view.js

```diff
@@ -1,89 +1,182 @@
+          td.className = 'highlighted';
+        }
 
-				tr.appendChild(td);
-			}
-			fragment.appendChild(tr);
-		}
-		removeChildren(this.sheetbodyEl);
-		this.sheetbodyEl.appendChild(fragment);
-	}
-	attachEventHandlers(){
-		this.sheetbodyEl.addEventListener('click',this.handleSheetClick.bind(this));
-		this.formulabarEl.addEventListener('keyup',this.handleFormulaBarChange.bind(this));
-	}
-	isColumnHeaderRow(row){
-		return row < 1;
-	}
-	handleFormulaBarChange(e){
-		const value = this.formulabarEl.value;
-		this.model.setValue(this.currentCellLocation,value);
-		this.renderTableBody();
-	}
-	handleSheetClick(e){
-		const col = e.target.cellIndex;
-		var row = e.target.parentElement.rowIndex;
+        tr.appendChild(td);
+      }
+      fragment.appendChild(tr);
+      if (this.checkLastRow(row)){
+        tr.className = 'last-row';
+      }
+    }
+    removeChildren(this.sheetbodyEl);
+    this.sheetbodyEl.appendChild(fragment);
+  }
+  highlightRow(colTarget,rowTarget){
+    let colTargetVal = colTarget;
+    let rowTargetVal = rowTarget;
+    const fragment = document.createDocumentFragment();
+    for(let row = 0; row < this.model.numRows; row++){
+      const tr = createTR();
+      tr.appendChild(createTH(row + 1));
+      for(let col = 0; col < this.model.numCols; col++){
+        this.currentCellLocation = {col: col, row: rowTargetVal-1};
+        const position = {col: col, row: row};
+        const value = this.model.getValue(position);
+        const td = createTD(value);
+        if(this.isCurrentCell(col,row) && !this.checkLastRow(row)){
+          td.className = 'highlighted';
+        }
 
-		if(!this.isColumnHeaderRow(row)){
-			row = row-1;
-			this.currentCellLocation = {col:col,row:row};
-			this.renderTableBody();
-		}
-		this.renderFormulaBar();
-	}
+        tr.appendChild(td);
+      }
+      fragment.appendChild(tr);
+      if (this.checkLastRow(row)){
+        tr.className = 'last-row';
+      }
+    }
+    removeChildren(this.sheetbodyEl);
+    this.sheetbodyEl.appendChild(fragment);
+  }
+  handleFormulaBarChange(e){
+    const value = this.formulabarEl.value;
+    this.model.setValue(this.currentCellLocation,value);
+    this.findSum(this.currentCellLocation);
+    this.renderTableBody();
+  }
+  findSum(currentCellLocation){
+    let value;
+    let sum = 0;
+    for(let row = 0; row < this.model.numRows-1; row++){
+      const position = {col: currentCellLocation.col, row: row};
+      value = this.model.getValue(position);
+      if(!Number.isNaN(parseInt(value))){
+        sum += parseInt(value);
+      }
+    }
+    const sumLocation = {col: currentCellLocation.col, row: this.model.numRows-1};
+    this.model.setValue(sumLocation,sum);
+  }
+  handleSheetClick(e){
+    let col = e.target.cellIndex;
+    let row = e.target.parentElement.rowIndex;
+    if(e.target.nodeName=='TD'){
+      if(!this.isColumnHeaderRow(row) || !this.isRowHeaderColumn(col)){
+        row = row-1; col = col-1;
```

Comment:
Give each of these two statements its own line. Again, this helps human readability when you scan down the file.

Also, you can abbreviate these as "row--;" and "col--:"